1.1/spec/18-Annex-C.adoc

```diff
@@ -1,6 +1,149 @@
+| Spatial Geometry (WKT)  | POINT(1 1) | http://www.opengis.net/ont/geosparql#wktLiteral[geo:wktLiteral] | "POINT(1 1)"^^geo:wktLiteral
+| Spatial Geometry (JSON)  | {"type": "Point", "coordinates":[1,1]} | http://www.opengis.net/ont/geosparql#geoJSONLiteral[geo:geoJSONLiteral] | "{"type": "Point", "coordinates":[1,1]}"^^geo:geoJSONLiteral
+| Temporal Literal  | 1969-07-20  1969-07-20T20:17:40Z | http://www.w3.org/2001/XMLSchema#date[xsd:date] http://www.w3.org/2001/XMLSchema#dateTime[xsd:dateTime] | "1969-07-20"^^xsd:date "1969-07-20T20:17:40Z"^^xsd:dateTime
+|======================
+
+===== C2.3 Property references
+
+CQL allows to reference properties in the Feature Collection it is targeting for filtering. A property reference is converted to a triple pattern as shown in the following example. A SPARQL variable ?item is assumed to represent the Feature Collection.
+
+|======================
+| Property Reference  | Triple pattern
+| name="OGC" | ?item my:name "OGC"^^xsd:string
+| number=5 | ?item my:number "5"^^xsd:integer
+| number>5 | ?item my:number ?number . FILTER(?number>5)
+|======================
+
+===== C2.4 Comparison Predicates
+
+CQL defines comparison predicates to compare two scalar expressions. A comparison predicate is converted to a triple pattern as shown in the following example. A SPARQL variable ?item is assumed to represent the Feature Collection.
+
+|======================
+| Comparison predicate  | Triple pattern | Comment
+| name="OGC" | ?item my:name "OGC"^^xsd:string | Equality statements can be converted to a triple pattern
+| number=5 | ?item my:number "5"^^xsd:integer | 
+| number>5 | ?item my:number ?number . FILTER(?number>5) | Arithmetic comparisons (<,>,>=,<=) are converted to filter expressions
+| number BETWEEN 5 AND 10 | ?item my:number ?number . FILTER(?number>=5 && ?number<=10) | BETWEEN statements are converted to arithmetic expressions
+| name IN ("OGC","W3C") | VALUES ?namevalues {"OGC" "W3C"} ?item my:name ?namevalues . | IN statements are expressed by VALUES statements
+| name IS NOT NULL | EXISTS {?item my:name ?name } | NOT NULL statements are converted to EXIST statements
+| name LIKE "OGC." | ?item my:name ?name . FILTER(regex(?name, "OGC.", "i" )) | LIKE statements are converted to SPARQL regex filters
+| INTERSECTS(geometry1, geometry2) | FILTER(http://www.opengis.net/def/function/geosparql/sfIntersects[geof:sfIntersects](?geometry1,?geometry2)) | The INTERSECTS filter statement is converted to a GeoSPARQL FILTER statement
+|======================
+
+- Filter CRS parameter equivalent does not yet exist in GeoSPARQL?
+
+===== C2.5 Geospatial Operators
+
+GeoSPARQL includes equivalents of many CQL filter functions as can be seen in the table below.
+
+|======================
+| CQL Filter Expression  | GeoSPARQL Filter Function 
+| CONTAINS(geometry1,POINT(1 1))  | FILTER(http://www.opengis.net/def/function/geosparql/sfContains[geof:sfContains](?geometry1,"POINT(1 1)"^^geo:wktLiteral))  
```

Comment:
Why have you used a literal instead of a variable for CONTAINS()? Can you just say `CONTAINS(geometry1,geometry2)` just like the others?

I don't think GeoSPARQL `sfContains()` is limited to a `POINT` for the second parameter.

Reply:
No it is not. It just followed the CQL Filter expression I used as an example. I could change it to use variables only, but the CQL Filter expressions do not allow the representation of variables as a second parameter. I wanted to express an equivalent representation of exactly the CQL statement.

Reply:
> the CQL Filter expressions do not allow the representation of variables as a second parameter

I didn't mean that: I wanted to make sure that the second parameter could be a POLYGON instead of a POINT. Hard coded as you have it makes it look like a user can only test for a POINT inside something (1st param) but it should be test for anything inside anything else.

Reply:
Ok, changed it to use the variable name instead